docs/guides/tutorials/chatgpt_bot.rst

```diff
@@ -0,0 +1,2316 @@
+      console.log("Embedding generation complete");
+    }
+
+    (async function main() {
+      await storeEmbeddings();
+    })();
+
+
+Running the script
+------------------
+
+Let's add a script to ``package.json`` that will invoke and execute
+``generate-embeddings.ts``.
+
+.. code-block:: json-diff
+
+      {
+        "name": "docs-chatbot",
+        "version": "0.1.0",
+        "private": true,
+        "scripts": {
+          "dev": "next dev",
+          "build": "next build",
+          "start": "next start",
+    -     "lint": "next lint"
+    +     "lint": "next lint",
+    +     "embeddings": "tsx generate-embeddings.ts"
+        },
+        "dependencies": {
+          "next": "^13.4.19",
+          "react": "18.2.0",
+          "react-dom": "18.2.0",
+          "sse.js": "^0.6.1",
+          "typescript": "5.1.6",
+          "edgedb": "^1.3.5",
+          "openai": "^4.0.1"
+        },
+        "devDependencies": {
+          "@types/node": "20.4.8",
+          "@types/react": "18.2.18",
+          "@types/react-dom": "18.2.7",
+          "@edgedb/generate": "^0.3.3",
+          "dotenv": "^16.3.1",
+          "gpt-tokenizer": "^2.1.1",
+          "tsx": "^3.12.7",
+          "autoprefixer": "10.4.14",
+          "eslint": "8.46.0",
+          "eslint-config-next": "13.4.13",
+          "postcss": "8.4.27",
+          "tailwindcss": "3.3.3"
+        }
+      }
+
+Now we can invoke the ``generate-embeddings.ts`` script from our terminal using
+a simple command:
+
+.. code-block:: bash
+
+   $ npm run embeddings
+
+After the script finishes, open the EdgeDB UI.
+
+.. code-block:: bash
+
+  $ edgedb ui
+
+Open your "edgedb" database and switch to the Data Explorer tab. You should see
+that the database has been updated with the embeddings and other relevant data.
+
+
+Answering user questions
+========================
+
+Now that we have the content's embeddings stored, we can start working on the
+handler for user questions. The user will submit a question to our server, and
+the handler will send them an answer back. We will define a route and an HTTP
+request handler for this task. Thanks to the power of Next.js, we can do all of
+this within our project using a `route handler`_.
+
+.. _route handler:
+  https://nextjs.org/docs/app/building-your-application/routing/route-handlers
+
+As we write our handler, one important consideration is that answers can be
+quite long. We could wait on the server side to get the whole answer from
+OpenAI and then send it to the client, but that would feel slow to the user.
+OpenAI supports streaming, so instead we can send the answer to the client in
+chunks, as they arrive to the server. With this approach, the user doesn't have
+to wait for the entire response before they start getting feedback and our API
+seems faster.
+
+In order to stream responses, we will use the browser's `server-sent events
+(SSE) API`_. Server-sent events enables a client to receive automatic updates
```

Comment:
I found this same sentence on Wikipedia: "Server-sent events enables a cli..."

But I think it's better to say that server maintains data transmissions instead initiates: 

`Server-sent events enables a client to receive automatic updates from a server via an HTTP connection, and describes how server maintains data transmissions to client once an initial client connection has been established. The client sends a request and with that request initiates a connection with our server. The server then sends data back to the client in chunks until all of the data is sent, at which point it closes the connection.`

Reply:
OK. I'll do a little work on that one and slot it in. Thank you!

Reply:
We should never do that? I believe that explanations on Wikipedia can be correct and the same as on the other places on more established websites like MDN. 
Not sure where I get this from when I firstly wrote it (now when I searched for better wording I found it there), but I never went through the guide the 2nd time so now I realised it can be better explained.

Reply:
Oh, we have to give credit to the organisations that came up with the text if we copy/paste it. Tnx

Reply:
Yeah, it's OK as long as we give attribution.